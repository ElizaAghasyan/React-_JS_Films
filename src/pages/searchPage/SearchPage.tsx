import {category, getMovieList, movieType, searchFilms} from "../../config/movieApi";
import {useEffect, useState} from "react";
import MovieList from "../../components/movieList/MovieList";
import Loading from "../../components/loading/Loading";
import { StateProperties } from "../home/Home";
import {useLocation} from "react-router-dom";

const styles = require('./SearchPage.module.scss');

const SearchPage = () => {
    const [searchValue, setSearchValue] = useState<StateProperties[]>([])
    const [page, setPage] = useState(1);

    const { search }  = useLocation()
    const query = new URLSearchParams(search)
    const movie = query.get(`movie`);

    useEffect(() => {
        const searchApi = async () => {
            if(!query) {
                return
            }
            let response;
            const params = {
                query: movie
            }
            try {
                response = await searchFilms(category.movie, {params})
                setSearchValue(response.data.results.splice(0,8));
                return response.data.results;
            }catch (e: any) {
                console.log(e)
            }
        }
        searchApi()
    }, [ movie ])

    const loadMore = async () => {
        const params = {
            page: page + 1
        };
        let  response = await getMovieList(movieType.upcoming, {params});
        setSearchValue([...searchValue, ...response.data.results]);
        setPage(page + 1)
    }

    return (
        <div className={styles.list}>
            <h1 className={styles.list_query}>Search results for {movie} </h1>
            <MovieList item={searchValue}/>
            <div className={styles.list_loadMore} onClick={loadMore}>
                <Loading/>
            </div>
        </div>
    )
}
export default SearchPage;
