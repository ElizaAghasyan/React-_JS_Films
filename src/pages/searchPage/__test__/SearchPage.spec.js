import {create} from "react-test-renderer";
import SearchPage from "../SearchPage";
import {createMemoryHistory} from "history";
import {BrowserRouter as Router} from "react-router-dom";

describe('Search Page', () => {
    const history = createMemoryHistory()

    test('Search page snapshot test', () => {
        const component = create(
            <Router history={history}>
                <SearchPage />
            </Router>

        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
