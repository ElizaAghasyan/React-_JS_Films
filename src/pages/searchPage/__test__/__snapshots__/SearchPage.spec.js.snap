// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Page Search page snapshot test 1`] = `
<div
  className="list"
>
  <h1
    className="list_query"
  >
    Search results for 
     
  </h1>
  <div
    className="cards"
  />
  <div
    className="list_loadMore"
    onClick={[Function]}
  >
    <div
      className="dots"
    >
      <div
        className="loading"
      >
        <button
          className="loading_btn"
        >
          Loading
        </button>
      </div>
    </div>
  </div>
</div>
`;
